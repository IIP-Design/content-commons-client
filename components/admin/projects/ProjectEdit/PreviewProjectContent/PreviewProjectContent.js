/**
 *
 * PreviewProjectContent
 *
 */

import React from 'react';
import { object, string } from 'prop-types';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import { Dropdown, Embed, Loader } from 'semantic-ui-react';

import DownloadVideo from 'components/admin/download/DownloadVideo/DownloadVideo';
import DownloadSrt from 'components/admin/download/DownloadSrt/DownloadSrt';
import DownloadThumbnail from 'components/admin/download/DownloadThumbnail/DownloadThumbnail';
import DownloadOtherFiles from 'components/admin/download/DownloadOtherFiles/DownloadOtherFiles';
import DownloadHelp from 'components/Video/DownloadHelp';

import ModalItem from 'components/modals/ModalItem/ModalItem';
import ModalContentMeta from 'components/modals/ModalContentMeta/ModalContentMeta';
import ModalDescription from 'components/modals/ModalDescription/ModalDescription';
import ModalPostMeta from 'components/modals/ModalPostMeta/ModalPostMeta';

import Notification from 'components/Notification/Notification';
import PopupTrigger from 'components/popups/PopupTrigger';
import PopupTabbed from 'components/popups/PopupTabbed';

import downloadIcon from 'static/icons/icon_download.svg';
import { getStreamData, getYouTubeId } from 'lib/utils';

import './PreviewProjectContent.scss';

/* eslint-disable react/prefer-stateless-function */
class PreviewProjectContent extends React.PureComponent {
  constructor( props ) {
    super( props );

    this.state = {
      dropDownIsOpen: false,
      selectedLanguage: 'English'
    };
  }

  getLanguages = units => (
    units.map( unit => ( {
      key: unit.language.languageCode,
      value: unit.language.displayName,
      text: unit.language.displayName
    } ) )
  )

  getProjectItems = units => (
    units.reduce( ( acc, unit ) => ( {
      ...acc,
      [unit.language.displayName]: unit
    } ), {} )
  );

  toggleArrow = () => {
    this.setState( prevState => ( {
      dropDownIsOpen: !prevState.dropDownIsOpen
    } ) );
  }

  selectLanguage = language => (
    this.setState( { selectedLanguage: language } )
  )

  handleChange = ( e, { value } ) => {
    this.toggleArrow();
    this.selectLanguage( value );
  }

  render() {
    const { error, loading, project } = this.props.data;

    if ( loading ) {
      return (
        <div
          className="preview-project-loader"
          style={ {
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            justifyContent: 'center',
            minHeight: '200px'
          } }
        >
          <Loader
            active
            inline="centered"
            style={ { marginBottom: '1em' } }
          />
          <p>Loading the project preview...</p>
        </div>
      );
    }

    if ( error ) return `Error! ${error.message}`;
    if ( !project || !Object.keys( project ).length ) return null;

    const {
      createdAt, updatedAt, projectType, team, units
    } = project;
    const { dropDownIsOpen, selectedLanguage } = this.state;

    const projectItems = this.getProjectItems( units );
    const selectedItem = projectItems[this.state.selectedLanguage];

    if ( !selectedItem || !Object.keys( selectedItem ).length ) return null;

    const {
      title, language, descPublic, files
    } = selectedItem;

    const currentUnit = files[0];
    const youTubeUrl = getStreamData( currentUnit.stream, 'youtube', 'embedUrl' );
    const vimeoUrl = getStreamData( currentUnit.stream, 'vimeo', 'embedUrl' );
    const { videoBurnedInStatus } = currentUnit;

    let thumbnailUrl = '';
    if ( selectedItem.thumbnails && selectedItem.thumbnails.length ) {
      thumbnailUrl = selectedItem.thumbnails[0].image.url;
    } else if ( project.thumbnails && project.thumbnails.length ) {
      thumbnailUrl = project.thumbnails[0].url;
    }

    const previewMsgStyles = {
      position: 'absolute',
      top: '0',
      left: '0',
      right: '0',
      // match Semantic UI border-radius
      borderTopLeftRadius: '0.28571429rem',
      borderTopRightRadius: '0.28571429rem',
      padding: '1em 1.5em',
      fontSize: '1em',
      backgroundColor: '#fdb81e'
    };

    return (
      <ModalItem
        className="project-preview"
        headline={ title }
        textDirection={ language.textDirection }
      >
        <Notification
          el="p"
          customStyles={ previewMsgStyles }
          msg={ `This is a preview of your ${projectType} project on Content Commons.` }
        />

        <div className="modal_options">
          <Dropdown
            className="modal_languages"
            value={ selectedLanguage }
            icon={ dropDownIsOpen ? 'chevron up' : 'chevron down' }
            options={ this.getLanguages( units ) }
            onClick={ this.toggleArrow }
            onChange={ this.handleChange }
          />

          <div className="trigger-container">
            <PopupTrigger
              toolTip="Download video"
              icon={ { img: downloadIcon, dim: 18 } }
              position="right"
              show={ projectType === 'video' }
              content={ (
                <PopupTabbed
                  title="Download this video."
                  panes={ [
                    {
                      title: 'Video File',
                      component: (
                        <DownloadVideo
                          selectedLanguageUnit={ selectedItem }
                          instructions={ `Download the video file in ${selectedLanguage}.
                            This download option is best for uploading this video to web pages.` }
                          burnedInCaptions={ videoBurnedInStatus === 'CAPTIONED' }
                        />
                      )
                    },
                    {
                      title: 'SRT',
                      component: (
                        <DownloadSrt
                          id={ this.props.id }
                          instructions="Download SRT(s)"
                        />
                      )
                    },
                    {
                      title: 'Thumbnail',
                      component: (
                        <DownloadThumbnail
                          id={ this.props.id }
                          instructions="Download Thumbnail(s)"
                        />
                      )
                    },
                    {
                      title: 'Other',
                      component: (
                        <DownloadOtherFiles
                          id={ this.props.id }
                          instructions="Download Other File(s)"
                        />
                      )
                    },
                    { title: 'Help', component: <DownloadHelp /> }
                  ] }
                />
              ) }
            />
          </div>
        </div>

        <div className="project-preview__content">
          { youTubeUrl
            && (
              <Embed
                id={ getYouTubeId( youTubeUrl ) }
                placeholder={ thumbnailUrl }
                source="youtube"
              />
            ) }

          { !youTubeUrl && vimeoUrl
            && (
              <Embed
                id={ getYouTubeId( youTubeUrl ) }
                placeholder={ thumbnailUrl }
                source="vimeo"
              />
            ) }

          <ModalContentMeta type={ projectType } dateUpdated={ updatedAt } />

          <ModalDescription description={ descPublic } />
        </div>

        <ModalPostMeta source={ team.name } datePublished={ createdAt } />
      </ModalItem>
    );
  }
}

PreviewProjectContent.propTypes = {
  id: string,
  data: object.isRequired
};

const VIDEO_PROJECT_PREVIEW_QUERY = gql`
  query VideoProjectPreview($id: ID!, $isReviewPage: Boolean!) {
    project: videoProject(id: $id) {
      id
      createdAt @skip(if: $isReviewPage)
      updatedAt @skip(if: $isReviewPage)
      projectType @skip(if: $isReviewPage)
      thumbnails @skip(if: $isReviewPage) {
        alt
        url
      }
      team @skip(if: $isReviewPage) {
        name
      }
      units {
        id
        title
        descPublic
        thumbnails {
          image {
            alt
            url
          }
        }
        language @skip(if: $isReviewPage) {
          languageCode
          displayName
          textDirection
        }
        files {
          id
          filename
          url
          filesize
          videoBurnedInStatus
          createdAt @include(if: $isReviewPage)
          duration @include(if: $isReviewPage)
          quality @include(if: $isReviewPage)
          dimensions {
            width
            height
          }
          stream {
            site
            embedUrl @skip(if: $isReviewPage)
            url @include(if: $isReviewPage)
          }
          language @include(if: $isReviewPage) {
            displayName
            textDirection
          }
        }
      }
    }
  }
`;

export default graphql( VIDEO_PROJECT_PREVIEW_QUERY, {
  options: props => ( {
    variables: {
      id: props.id,
      isReviewPage: false
    },
  } )
} )( PreviewProjectContent );

export { VIDEO_PROJECT_PREVIEW_QUERY };
