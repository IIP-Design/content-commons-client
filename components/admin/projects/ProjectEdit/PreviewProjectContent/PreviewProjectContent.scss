.project-preview {
  .modal_headline {
    margin-top: 1.5em;
  }
}

.ui.dimmer.modals {
  .preview-project-loader {
    .ui.loader:before {
      border-color: rgba(0, 0, 0, 0.1);
    }
    .ui.loader:after {
      border-color: #767676 transparent transparent;
    }
  }
}
