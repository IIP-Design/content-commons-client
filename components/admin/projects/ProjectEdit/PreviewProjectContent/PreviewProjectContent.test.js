import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import { videoProjectPreview } from 'components/admin/projects/ProjectEdit/mockData';
import PreviewProjectContent, { VIDEO_PROJECT_PREVIEW_QUERY } from './PreviewProjectContent';

const props = { id: '123' };
const mocks = [
  {
    request: {
      query: VIDEO_PROJECT_PREVIEW_QUERY,
      variables: { id: '123', isReviewPage: false }
    },
    result: {
      data: { project: videoProjectPreview }
    }
  }
];

const Component = (
  <MockedProvider mocks={ mocks } addTypename={ false }>
    <PreviewProjectContent { ...props } />
  </MockedProvider>
);

describe( '<PreviewProjectContent />', () => {
  it( 'renders initial loading state without crashing', () => {
    const wrapper = mount( Component );
    const preview = wrapper.find( 'PreviewProjectContent' );

    expect( preview.exists() ).toEqual( true );
    expect( preview.find( '.preview-project-loader' ).exists() )
      .toEqual( true );
    expect( preview.find( 'Loader' ).exists() )
      .toEqual( true );
    expect( toJSON( preview ) ).toMatchSnapshot();
  } );

  it( 'renders final state without crashing', async () => {
    const wrapper = mount( Component );
    // wait for the data and !loading
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );

    expect( preview.exists() ).toEqual( true );
    expect( preview.find( '.preview-project-loader' ).exists() )
      .toEqual( false );
    expect( preview.find( 'Loader' ).exists() )
      .toEqual( false );
    expect( preview.find( 'ModalItem.project-preview' ).exists() )
      .toEqual( true );
    expect( toJSON( preview ) ).toMatchSnapshot();
  } );

  it( 'renders error message if error is thrown', async () => {
    const errorMocks = [
      {
        request: {
          query: VIDEO_PROJECT_PREVIEW_QUERY,
          variables: { id: '123', isReviewPage: false }
        },
        result: {
          errors: [{ message: 'There was an error.' }]
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={ errorMocks } addTypename={ false }>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );

    expect( preview.text() )
      .toEqual( 'Error! GraphQL error: There was an error.' );
  } );


  it( 'renders `null` if `!project`', async () => {
    const nullMocks = [
      {
        request: {
          query: VIDEO_PROJECT_PREVIEW_QUERY,
          variables: { id: '123', isReviewPage: false }
        },
        result: {
          data: { project: null }
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={ nullMocks } addTypename={ false }>
        <PreviewProjectContent { ...props } />
      </MockedProvider>
    );

    // wait for the data and !loading
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );

    expect( preview.html() ).toEqual( null );
  } );

  it( 'sets `dropDownIsOpen` and `selectedLanguage` in initial `state`', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const { dropDownIsOpen, selectedLanguage } = inst.state;

    expect( dropDownIsOpen ).toEqual( false );
    expect( selectedLanguage ).toEqual( 'English' );
  } );

  it( '`getLanguages` returns an array of language objects', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'getLanguages' );
    const { units } = inst.props.data.project;
    const languages = inst.getLanguages( units );

    expect( spy ).toHaveBeenCalled();
    expect( languages ).toEqual( [
      { key: 'en', value: 'English', text: 'English' },
      { key: 'fr', value: 'French', text: 'French' }
    ] );
  } );

  it( '`getProjectItems` sets `projectItems` in initial `state`', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'getProjectItems' );
    const { units } = inst.props.data.project;
    const projectItems = inst.getProjectItems( units );
    const languages = units.map( unit => unit.language.displayName );

    expect( spy ).toHaveBeenCalled();
    languages.forEach( ( language, i ) => {
      expect( projectItems[language] ).toEqual( units[i] );
    } );
  } );

  it( '`toggleArrow` updates `dropDownIsOpen` in state', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'toggleArrow' );
    const dropDownIsOpen = () => inst.state.dropDownIsOpen;

    expect( dropDownIsOpen() ).toEqual( false );

    inst.toggleArrow();
    expect( spy ).toHaveBeenCalledTimes( 1 );
    expect( dropDownIsOpen() ).toEqual( true );

    inst.toggleArrow();
    expect( spy ).toHaveBeenCalledTimes( 2 );
    expect( dropDownIsOpen() ).toEqual( false );
  } );

  it( '`selectLanguage` updates `selectedLanguage` in state', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const spy = jest.spyOn( inst, 'selectLanguage' );
    const selectedLanguage = () => inst.state.selectedLanguage;

    expect( selectedLanguage() ).toEqual( 'English' );

    inst.selectLanguage( 'French' );
    expect( spy ).toHaveBeenCalledTimes( 1 );
    expect( selectedLanguage() ).toEqual( 'French' );

    inst.selectLanguage( 'English' );
    expect( spy ).toHaveBeenCalledTimes( 2 );
    expect( selectedLanguage() ).toEqual( 'English' );
  } );


  it( '`handleChange` calls `toggleArrow` and `selectLanguage`', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const preview = wrapper.find( 'PreviewProjectContent' );
    const inst = preview.instance();
    const selectedLanguage = () => inst.state.selectedLanguage;

    const handleChangeSpy = jest.spyOn( inst, 'handleChange' );
    const toggleArrowSpy = jest.spyOn( inst, 'toggleArrow' );
    const selectLanguageSpy = jest.spyOn( inst, 'selectLanguage' );

    const e = {};
    const selection = { value: 'French' };

    expect( selectedLanguage() ).toEqual( 'English' );

    inst.handleChange( e, selection );

    expect( handleChangeSpy ).toHaveBeenCalledWith( e, selection );
    expect( toggleArrowSpy ).toHaveBeenCalledTimes( 1 );
    expect( selectLanguageSpy ).toHaveBeenCalledWith( selection.value );
    expect( selectedLanguage() ).toEqual( selection.value );

    selection.value = 'English';
    inst.handleChange( e, selection );

    expect( handleChangeSpy ).toHaveBeenCalledWith( e, selection );
    expect( handleChangeSpy ).toHaveBeenCalledTimes( 2 );
    expect( toggleArrowSpy ).toHaveBeenCalledTimes( 2 );
    expect( selectLanguageSpy ).toHaveBeenCalledWith( selection.value );
    expect( selectLanguageSpy ).toHaveBeenCalledTimes( 2 );
    expect( selectedLanguage() ).toEqual( selection.value );
  } );
} );
