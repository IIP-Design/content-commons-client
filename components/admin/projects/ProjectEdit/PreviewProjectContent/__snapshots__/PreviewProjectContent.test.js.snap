// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewProjectContent /> renders final state without crashing 1`] = `
<PreviewProjectContent
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": false,
      "networkStatus": 7,
      "project": Object {
        "createdAt": "2019-03-20T15:09:24.975Z",
        "id": "123",
        "projectType": "video",
        "team": Object {
          "name": "IIP Video Production",
        },
        "thumbnails": Array [
          Object {
            "alt": "A man wearing a hardhat walks through an empty factory.",
            "url": "https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg",
          },
        ],
        "units": Array [
          Object {
            "descPublic": "The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today.",
            "files": Array [
              Object {
                "dimensions": Object {
                  "height": 1080,
                  "width": 1920,
                },
                "filename": "madeinamerica_english_asdkjaf_al;kdflkeori_erpoiuzx,mnvz.mp4",
                "filesize": 662595174,
                "id": "cjsw7urxd00oc0756llfh4ihs",
                "stream": Object {
                  "embedUrl": "https://www.youtube.com/embed/1evw4fRu3bo",
                  "site": "YouTube",
                },
                "url": "https://video-download-url.com",
                "videoBurnedInStatus": "CLEAN",
              },
            ],
            "id": "cjsx8fj5h01ez0756cd0t6r30",
            "language": Object {
              "displayName": "English",
              "languageCode": "en",
              "textDirection": "LTR",
            },
            "thumbnails": Array [
              Object {
                "image": Object {
                  "alt": "A man wearing a hardhat walks through an empty factory.",
                  "url": "https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg",
                },
              },
            ],
            "title": "Made in America",
          },
          Object {
            "descPublic": "La valeur et le sens de l’expression Fabriqué en Amérique, ont leur origine dans un passé riche d’innovation et de persévérance. Découvrez une partie de l’histoire de l’industrie manufacturière aux États-Unis et l’influence positive qu’elle a eue sur le monde, dans le passé et encore aujourd’hui.",
            "files": Array [
              Object {
                "dimensions": Object {
                  "height": 1080,
                  "width": 1920,
                },
                "filename": "madeinamerica_french.mp4",
                "filesize": 662595174,
                "id": "cjsxe6k6w01z7075657wt2ork",
                "stream": Object {
                  "embedUrl": "https://www.youtube.com/embed/1evw4fRu3bo",
                  "site": "YouTube",
                },
                "url": "https://video-download-url.com",
                "videoBurnedInStatus": "CLEAN",
              },
            ],
            "id": "cjt01xotk02fu0756743g45a8",
            "language": Object {
              "displayName": "French",
              "languageCode": "fr",
              "textDirection": "LTR",
            },
            "thumbnails": Array [
              Object {
                "image": Object {
                  "alt": "A man wearing a hardhat walks through an empty factory.",
                  "url": "https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg",
                },
              },
            ],
            "title": "Fabriqué en Amérique",
          },
        ],
        "updatedAt": "2019-04-02T16:28:31.888Z",
      },
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
        "isReviewPage": false,
      },
    }
  }
  id="123"
>
  <ModalItem
    className="project-preview"
    headline="Made in America"
    textDirection="LTR"
  >
    <div
      className="modal LTR project-preview"
    >
      <h1
        className="modal_headline"
      >
        Made in America
      </h1>
      <Notification
        customStyles={
          Object {
            "backgroundColor": "#fdb81e",
            "borderTopLeftRadius": "0.28571429rem",
            "borderTopRightRadius": "0.28571429rem",
            "fontSize": "1em",
            "left": "0",
            "padding": "1em 1.5em",
            "position": "absolute",
            "right": "0",
            "top": "0",
          }
        }
        el="p"
        msg="This is a preview of your video project on Content Commons."
      >
        <p
          style={
            Object {
              "backgroundColor": "#fdb81e",
              "borderTopLeftRadius": "0.28571429rem",
              "borderTopRightRadius": "0.28571429rem",
              "fontSize": "1em",
              "left": "0",
              "padding": "1em 1.5em",
              "position": "absolute",
              "right": "0",
              "top": "0",
            }
          }
        >
          This is a preview of your video project on Content Commons.
        </p>
      </Notification>
      <div
        className="modal_options"
      >
        <Dropdown
          additionLabel="Add "
          additionPosition="top"
          className="modal_languages"
          closeOnBlur={true}
          deburr={false}
          icon="chevron down"
          minCharacters={1}
          noResultsMessage="No results found."
          onChange={[Function]}
          onClick={[Function]}
          openOnFocus={true}
          options={
            Array [
              Object {
                "key": "en",
                "text": "English",
                "value": "English",
              },
              Object {
                "key": "fr",
                "text": "French",
                "value": "French",
              },
            ]
          }
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          value="English"
          wrapSelection={true}
        >
          <div
            aria-expanded={false}
            className="ui dropdown modal_languages"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <div
              aria-live="polite"
              className="text"
              role="alert"
            >
              English
            </div>
            <Icon
              as="i"
              className=""
              name="chevron down"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="chevron down icon"
                onClick={[Function]}
              />
            </Icon>
            <DropdownMenu>
              <div
                className="menu transition"
              >
                <DropdownItem
                  active={true}
                  key="en"
                  onClick={[Function]}
                  selected={true}
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                  text="English"
                  value="English"
                >
                  <div
                    aria-checked={true}
                    aria-selected={true}
                    className="active selected item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                  >
                    <span
                      className="text"
                    >
                      English
                    </span>
                  </div>
                </DropdownItem>
                <DropdownItem
                  active={false}
                  key="fr"
                  onClick={[Function]}
                  selected={false}
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                  text="French"
                  value="French"
                >
                  <div
                    aria-checked={false}
                    aria-selected={false}
                    className="item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                  >
                    <span
                      className="text"
                    >
                      French
                    </span>
                  </div>
                </DropdownItem>
              </div>
            </DropdownMenu>
          </div>
        </Dropdown>
        <div
          className="trigger-container"
        >
          <PopupTrigger
            content={
              <PopupTabbed
                panes={
                  Array [
                    Object {
                      "component": <DownloadVideo
                        burnedInCaptions={false}
                        instructions="Download the video file in English.
                            This download option is best for uploading this video to web pages."
                        selectedLanguageUnit={
                          Object {
                            "descPublic": "The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today.",
                            "files": Array [
                              Object {
                                "dimensions": Object {
                                  "height": 1080,
                                  "width": 1920,
                                },
                                "filename": "madeinamerica_english_asdkjaf_al;kdflkeori_erpoiuzx,mnvz.mp4",
                                "filesize": 662595174,
                                "id": "cjsw7urxd00oc0756llfh4ihs",
                                "stream": Object {
                                  "embedUrl": "https://www.youtube.com/embed/1evw4fRu3bo",
                                  "site": "YouTube",
                                },
                                "url": "https://video-download-url.com",
                                "videoBurnedInStatus": "CLEAN",
                              },
                            ],
                            "id": "cjsx8fj5h01ez0756cd0t6r30",
                            "language": Object {
                              "displayName": "English",
                              "languageCode": "en",
                              "textDirection": "LTR",
                            },
                            "thumbnails": Array [
                              Object {
                                "image": Object {
                                  "alt": "A man wearing a hardhat walks through an empty factory.",
                                  "url": "https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg",
                                },
                              },
                            ],
                            "title": "Made in America",
                          }
                        }
                      />,
                      "title": "Video File",
                    },
                    Object {
                      "component": <Apollo(DownloadSrt)
                        id="123"
                        instructions="Download SRT(s)"
                      />,
                      "title": "SRT",
                    },
                    Object {
                      "component": <Apollo(DownloadThumbnail)
                        id="123"
                        instructions="Download Thumbnail(s)"
                      />,
                      "title": "Thumbnail",
                    },
                    Object {
                      "component": <Apollo(DownloadOtherFiles)
                        id="123"
                        instructions="Download Other File(s)"
                      />,
                      "title": "Other",
                    },
                    Object {
                      "component": <DownloadHelp />,
                      "title": "Help",
                    },
                  ]
                }
                title="Download this video."
              />
            }
            icon={
              Object {
                "dim": 18,
                "img": Object {},
              }
            }
            position="right"
            show={true}
            toolTip="Download video"
          >
            <span
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <Popup
                className="popupElem_wrapper"
                content={
                  <PopupTabbed
                    panes={
                      Array [
                        Object {
                          "component": <DownloadVideo
                            burnedInCaptions={false}
                            instructions="Download the video file in English.
                            This download option is best for uploading this video to web pages."
                            selectedLanguageUnit={
                              Object {
                                "descPublic": "The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today.",
                                "files": Array [
                                  Object {
                                    "dimensions": Object {
                                      "height": 1080,
                                      "width": 1920,
                                    },
                                    "filename": "madeinamerica_english_asdkjaf_al;kdflkeori_erpoiuzx,mnvz.mp4",
                                    "filesize": 662595174,
                                    "id": "cjsw7urxd00oc0756llfh4ihs",
                                    "stream": Object {
                                      "embedUrl": "https://www.youtube.com/embed/1evw4fRu3bo",
                                      "site": "YouTube",
                                    },
                                    "url": "https://video-download-url.com",
                                    "videoBurnedInStatus": "CLEAN",
                                  },
                                ],
                                "id": "cjsx8fj5h01ez0756cd0t6r30",
                                "language": Object {
                                  "displayName": "English",
                                  "languageCode": "en",
                                  "textDirection": "LTR",
                                },
                                "thumbnails": Array [
                                  Object {
                                    "image": Object {
                                      "alt": "A man wearing a hardhat walks through an empty factory.",
                                      "url": "https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg",
                                    },
                                  },
                                ],
                                "title": "Made in America",
                              }
                            }
                          />,
                          "title": "Video File",
                        },
                        Object {
                          "component": <Apollo(DownloadSrt)
                            id="123"
                            instructions="Download SRT(s)"
                          />,
                          "title": "SRT",
                        },
                        Object {
                          "component": <Apollo(DownloadThumbnail)
                            id="123"
                            instructions="Download Thumbnail(s)"
                          />,
                          "title": "Thumbnail",
                        },
                        Object {
                          "component": <Apollo(DownloadOtherFiles)
                            id="123"
                            instructions="Download Other File(s)"
                          />,
                          "title": "Other",
                        },
                        Object {
                          "component": <DownloadHelp />,
                          "title": "Help",
                        },
                      ]
                    }
                    title="Download this video."
                  />
                }
                horizontalOffset={4}
                keepInViewPort={true}
                on="click"
                position="bottom right"
                trigger={
                  <Button
                    as="button"
                    className="trigger"
                    tooltip="Download video"
                  >
                    <img
                      alt="Download video"
                      height={18}
                      src={Object {}}
                      width={18}
                    />
                     
                  </Button>
                }
              >
                <Portal
                  closeOnDocumentClick={true}
                  closeOnEscape={true}
                  closeOnTriggerClick={true}
                  eventPool="default"
                  onClose={[Function]}
                  onMount={[Function]}
                  onOpen={[Function]}
                  onUnmount={[Function]}
                  openOnTriggerClick={true}
                  trigger={
                    <Button
                      as="button"
                      className="trigger"
                      tooltip="Download video"
                    >
                      <img
                        alt="Download video"
                        height={18}
                        src={Object {}}
                        width={18}
                      />
                       
                    </Button>
                  }
                  triggerRef={[Function]}
                >
                  <Ref
                    innerRef={[Function]}
                  >
                    <Button
                      as="button"
                      className="trigger"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      tooltip="Download video"
                    >
                      <button
                        className="ui button trigger"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        tooltip="Download video"
                      >
                        <img
                          alt="Download video"
                          height={18}
                          src={Object {}}
                          width={18}
                        />
                         
                      </button>
                    </Button>
                  </Ref>
                </Portal>
              </Popup>
            </span>
          </PopupTrigger>
        </div>
      </div>
      <div
        className="project-preview__content"
      >
        <Embed
          id={null}
          placeholder="https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg"
          source="youtube"
        >
          <div
            className="ui embed"
            onClick={[Function]}
          >
            <Icon
              as="i"
              name="video play"
            >
              <i
                aria-hidden="true"
                className="video play icon"
              />
            </Icon>
            <img
              className="placeholder"
              src="https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg"
            />
          </div>
        </Embed>
        <ModalContentMeta
          dateUpdated="2019-04-02T16:28:31.888Z"
          type="video"
        >
          <section
            className="modal_section modal_section--metaContent"
          >
            <div
              className="modal_meta_wrapper"
            >
              <div
                className="modal_meta"
              >
                <span
                  className="modal_meta_content modal_meta_content--filetype"
                >
                  File Type: video
                </span>
                <span
                  className="modal_meta_content modal_meta_content--date"
                >
                  Updated: April 02, 2019
                </span>
              </div>
            </div>
          </section>
        </ModalContentMeta>
        <ModalDescription
          description="The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today."
        >
          <section
            className="modal_section modal_section--description"
          >
            <p
              className="modal_description_text"
            >
              The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today.
            </p>
          </section>
        </ModalDescription>
      </div>
      <ModalPostMeta
        datePublished="2019-03-20T15:09:24.975Z"
        source="IIP Video Production"
      >
        <section
          className="modal_section modal_section--postMeta"
        >
          <span
            className="modal_postmeta_content"
          >
            Source:
             
            IIP Video Production
          </span>
          <span
            className="modal_postmeta_content"
          >
            Date Published: March 20, 2019
          </span>
        </section>
      </ModalPostMeta>
    </div>
  </ModalItem>
</PreviewProjectContent>
`;

exports[`<PreviewProjectContent /> renders initial loading state without crashing 1`] = `
<PreviewProjectContent
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": true,
      "networkStatus": 1,
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
        "isReviewPage": false,
      },
    }
  }
  id="123"
>
  <div
    className="preview-project-loader"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": "200px",
      }
    }
  >
    <Loader
      active={true}
      inline="centered"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="ui active centered inline loader"
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      />
    </Loader>
    <p>
      Loading the project preview...
    </p>
  </div>
</PreviewProjectContent>
`;
